server/athenian/api/auth.py

```diff
@@ -0,0 +1,119 @@
+            raise AuthError("Authorization header is expected", 401)
+
+        parts = auth.split()
+
+        if parts[0].lower() != "bearer":
+            raise AuthError("Authorization header must start with Bearer", 401)
+        elif len(parts) == 1:
+            raise AuthError("Token not found", 401)
+        elif len(parts) > 2:
+            raise AuthError("Authorization header must be Bearer token", 401)
+
+        token = parts[1]
+        return token
+
+    def _get_user_info(self, token):
+        # TODO: cache based on decoded claims
+        jsonurl = urlopen(HTTPRequest("https://"+self._domain+"/userinfo", headers={"Authorization": "Bearer "+token}))
+        resp = json.loads(jsonurl.read())
+        return User(resp)
+
+    @aiohttp.web.middleware
+    async def middleware(self, request, handler):
+        if self._check_request(request):
+            return await handler(request)
+
+        # TODO: update periodically instead of pulling it on each request
+        jsonurl = urlopen("https://"+self._domain+"/.well-known/jwks.json")
+        jwks = json.loads(jsonurl.read())
+
+        try:
+            token = self._get_token_auth_header(request)
+            unverified_header = jwt.get_unverified_header(token)
+        except AuthError as e:
+            return aiohttp.web.Response(body=e.error, status=e.status_code)
+        except jwt.JWTError:
+            return aiohttp.web.Response(body="Invalid header. Use an RS256 signed JWT Access Token.", status=401)
+        if unverified_header["alg"] != self._algos[0]:
+            return aiohttp.web.Response(body="Invalid header. Use an RS256 signed JWT Access Token.", status=401)
+
+        # There might be multiple keys, find the one used to sign this request
+        rsa_key = None
+        for key in jwks["keys"]:
+            if key["kid"] == unverified_header["kid"]:
+                rsa_key = {
+                    "kty": key["kty"],
+                    "kid": key["kid"],
+                    "use": key["use"],
+                    "n": key["n"],
+                    "e": key["e"]
+                }
+
+        if rsa_key is None:
+            return aiohttp.web.Response(body="Unable to find appropriate key", status=401)
+
+        try:
+            payload = jwt.decode(
+                token,
+                rsa_key,
+                algorithms=self._algos,
+                audience=self._audience,
+                issuer="https://"+self._domain+"/"
+            )
+        except jwt.ExpiredSignatureError:
+            return aiohttp.web.Response(body="token expired", status=401)
+        except jwt.JWTClaimsError:
+            return aiohttp.web.Response(body="incorrect claims, please check the audience and issuer", status=401)
+        except Exception:
+            return aiohttp.web.Response(body="Unable to parse authentication token.", status=401)
+
+        user = self._get_user_info(token)
+        logging.getLogger(__package__).info("User %s", user.username)
```

Comment:
```suggestion
        logging.getLogger("auth").info("User %s", user.username)
```
So that we can silence the auth logs or make them more important.

Reply:
Is there a way to attach the name to `aiohttp` logs, btw?

Reply:
The only not-too-magical way I see is to wrap the logging calls so that they always require setting the user info.